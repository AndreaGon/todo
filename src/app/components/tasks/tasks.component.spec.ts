import {ComponentFixture, TestBed} from '@angular/core/testing';
import {BrowserModule, By} from '@angular/platform-browser';
import {HttpClientTestingModule} from '@angular/common/http/testing';

import {DebugElement, CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';

import {TasksComponent} from './tasks.component';
import {TaskItemComponent} from '../task-item/task-item.component';
//import {TaskService} from '../../services/task.service';

describe('TasksComponent', ()=>{
  let component: TasksComponent;
  let fixture: ComponentFixture<TasksComponent>;
  let debugging: DebugElement;
  let element: HTMLElement;

  beforeEach(async()=>{
    await TestBed.configureTestingModule({
      imports:[BrowserModule, HttpClientTestingModule],
      declarations: [TasksComponent, TaskItemComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    })
    .compileComponents().then(()=>{
      fixture = TestBed.createComponent(TasksComponent);

      component = fixture.componentInstance;

      fixture.detectChanges();
    });
  });

  //Demo of snapshot test for Jest
  it(`should display a task item`, async()=>{
    component.tasks = [
      {
        id: 1,
        text: "Testing title",
        day: "Testing day",
        reminder: false
      }
    ]

    component.ngOnInit();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it(`should display a new data in the task list when there is new data`, async()=>{
    component.tasks = [
      {
        id: 1,
        text: "Testing title",
        day: "Testing day",
        reminder: false
      }
    ]

    component.ngOnInit();
    fixture.detectChanges();

    //Test if there is ONE data in the list UI
    expect(fixture).toMatchSnapshot();

    component.tasks.push(
      {
        id: 2,
        text: "Testing title 2",
        day: "Testing day 2",
        reminder: true
      }
    );

    fixture.detectChanges();

    //Test if there is TWO data in the list UI
    expect(fixture).toMatchSnapshot();

  });

  it(`should not display a task item if empty`, async()=>{
    component.tasks = [];
    component.ngOnInit();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it(`task should have green marker when set reminder is true`, async()=>{
    component.tasks = [
      {
        id: 1,
        text: "Testing title",
        day: "Testing day",
        reminder: false
      },
      {
        id: 2,
        text: "Testing title",
        day: "Testing day",
        reminder: true
      },
      {
        id: 3,
        text: "Testing title",
        day: "Testing day",
        reminder: true
      },
      {
        id: 4,
        text: "Testing title",
        day: "Testing day",
        reminder: true
      },
      {
        id: 5,
        text: "Testing title",
        day: "Testing day",
        reminder: false
      }
      
    ]

    component.ngOnInit();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

});