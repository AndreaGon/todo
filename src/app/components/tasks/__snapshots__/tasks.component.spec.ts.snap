// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksComponent should display a new data in the task list when there is new data 1`] = `
<app-tasks
  taskService={[Function TaskService]}
  tasks={[Function Array]}
>
  <app-add-task /><app-task-item>
    <div
      class="task"
    >
      <h3>
        Testing title 
        <fa-icon
          style="color: red;"
        />
      </h3>
      <p>
        Testing day
      </p>
    </div>
  </app-task-item>
</app-tasks>
`;

exports[`TasksComponent should display a new data in the task list when there is new data 2`] = `
<app-tasks
  taskService={[Function TaskService]}
  tasks={[Function Array]}
>
  <app-add-task /><app-task-item>
    <div
      class="task"
    >
      <h3>
        Testing title 
        <fa-icon
          style="color: red;"
        />
      </h3>
      <p>
        Testing day
      </p>
    </div>
  </app-task-item><app-task-item>
    <div
      class="task reminder"
    >
      <h3>
        Testing title 2 
        <fa-icon
          style="color: red;"
        />
      </h3>
      <p>
        Testing day 2
      </p>
    </div>
  </app-task-item>
</app-tasks>
`;

exports[`TasksComponent should display a task item 1`] = `
<app-tasks
  taskService={[Function TaskService]}
  tasks={[Function Array]}
>
  <app-add-task /><app-task-item>
    <div
      class="task"
    >
      <h3>
        Testing title 
        <fa-icon
          style="color: red;"
        />
      </h3>
      <p>
        Testing day
      </p>
    </div>
  </app-task-item>
</app-tasks>
`;

exports[`TasksComponent should not display a task item if empty 1`] = `
<app-tasks
  taskService={[Function TaskService]}
  tasks={[Function Array]}
>
  <app-add-task />
</app-tasks>
`;

exports[`TasksComponent task should have green marker when set reminder is true 1`] = `
<app-tasks
  taskService={[Function TaskService]}
  tasks={[Function Array]}
>
  <app-add-task /><app-task-item>
    <div
      class="task"
    >
      <h3>
        Testing title 
        <fa-icon
          style="color: red;"
        />
      </h3>
      <p>
        Testing day
      </p>
    </div>
  </app-task-item><app-task-item>
    <div
      class="task reminder"
    >
      <h3>
        Testing title 
        <fa-icon
          style="color: red;"
        />
      </h3>
      <p>
        Testing day
      </p>
    </div>
  </app-task-item><app-task-item>
    <div
      class="task reminder"
    >
      <h3>
        Testing title 
        <fa-icon
          style="color: red;"
        />
      </h3>
      <p>
        Testing day
      </p>
    </div>
  </app-task-item><app-task-item>
    <div
      class="task reminder"
    >
      <h3>
        Testing title 
        <fa-icon
          style="color: red;"
        />
      </h3>
      <p>
        Testing day
      </p>
    </div>
  </app-task-item><app-task-item>
    <div
      class="task"
    >
      <h3>
        Testing title 
        <fa-icon
          style="color: red;"
        />
      </h3>
      <p>
        Testing day
      </p>
    </div>
  </app-task-item>
</app-tasks>
`;
